### PR TITLE
Update documentation and config files to add WandB details

### PR DESCRIPTION
**Pull request by naga-karthik**
#### GitHub

- [X] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [X] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [X] I've assigned a reviewer

#### PR contents

- [X] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [X] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

This PR brings an updated version of the documentation adding the relevant details regarding the WandB parameters in the configuration files. I also checked the updated documentation locally by doing a quick `make html` to build the docs. Additionally, all the config files in the tutorials have also been updated with the "wandb" key.

### CODE REVIEW

File: docs/source/configuration_file.rst [hunk 1]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
...
+            "the groups can be found on clicking the ``Projects`` tab on the left."
+        ],
+        "type": "string"
+    }
+
+.. code-block:: JSON
+
+    {
+        "wandb": {
+            "group_name": "my-temp-group"
```
Comment (kiristern) [inline docs/source/configuration_file.rst:226]: not sure I understand difference between "title" and "group_name".

File: docs/source/configuration_file.rst [hunk 2]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
...
+
+.. note::
+    There are two important points to be noted: 
+    (1) Gradients can be large so they can consume more storage space if ``log_grads_every`` is set to a small number, 
+    (2) ``log_grads_every`` also depends on the total duration of training, i.e. if the model is run for only
+    a few epochs, gradients might not be logged if ``log_grads_every`` is too large. Hence, the right frequency of
+    gradient logging depends on the training duration and model size.
+
+.. note::
+    If ``debug = True`` is specified in the config file, the training and validation input images, ground truth labels, and
```
Comment (kanishk16) [inline docs/source/configuration_file.rst:280]: I guess you meant `debugging`?

File: docs/source/configuration_file.rst [hunk 3]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
...
+    }
+
+
+.. jsonschema::
+
+    {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "title": "group_name",
+        "$$description": [
+            "This defines the name of the group to which the runs will be synced. On the WandB Dashboard,\n",
```
Comment (kanishk16) [inline docs/source/configuration_file.rst:216]: Same as above

File: docs/source/configuration_file.rst [hunk 4]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
...
+    }
+
+
+.. jsonschema::
+
+    {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "title": "run_name",
+        "$$description": [
+            "This defines the name of the current run (or, experiment). All the previous and active runs\n",
```
Comment (kanishk16) [inline docs/source/configuration_file.rst:237]: Same as above

File: docs/source/configuration_file.rst [hunk 5]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
...
+without an account, since the metrics are logged on Tensorboard by default. 
+
+
+.. jsonschema::
+
+    {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "title": "wandb_api_key",
+        "$$description": [
+            "This is a private key used to sync the local wandb folder with the wandb dashboard accessible through the browser.\n",
```
Comment (kanishk16) [inline docs/source/configuration_file.rst:172]:
```suggestion
            "A private key used to sync the local wandb folder with the wandb dashboard accessible through the browser.\n",
```

File: docs/source/configuration_file.rst [hunk 6]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
...
+            "The API key can be found from the browser in your WandB Account's Settings, under the section ``API Keys``.\n",
+            "Note that once it is successfully authorized for the first time, a message will be printed in the terminal notifying\n",
+            "that the API key is stored in the ``.netrc`` file in the ``/home`` folder. From here on, this key-value pair in the\n",
```
Comment (kanishk16) [inline docs/source/configuration_file.rst:175]:
```suggestion
            "that the API key is stored in the ``.netrc`` file in the ``/home`` folder. From then on, this key-value pair in the\n",
```

File: docs/source/configuration_file.rst [hunk 7]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
     }
 
 
+Weights and Biases (WandB)
```
Comment (kanishk16) [inline docs/source/configuration_file.rst:156]:
```suggestion
Weights & Biases (WandB)
```

File: docs/source/configuration_file.rst [hunk 8]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
...
+
+.. jsonschema::
+
+    {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "title": "wandb_api_key",
+        "$$description": [
+            "This is a private key used to sync the local wandb folder with the wandb dashboard accessible through the browser.\n",
+            "The API key can be found from the browser in your WandB Account's Settings, under the section ``API Keys``.\n",
+            "Note that once it is successfully authorized for the first time, a message will be printed in the terminal notifying\n",
```
Comment (kanishk16) [inline docs/source/configuration_file.rst:174]:
```suggestion
            "Note that once it is successfully authenticated, a message would be printed in the terminal notifying\n",
```

File: docs/source/configuration_file.rst [hunk 9]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
...
+            "that the API key is stored in the ``.netrc`` file in the ``/home`` folder. From here on, this key-value pair in the\n",
+            "config file can be left empty, like ``'wandb_api_key': ''``"
```
Comment (kanishk16) [inline docs/source/configuration_file.rst:176]:
```suggestion
            "config file could be omitted, like ``'wandb_api_key': ''``"
```

File: docs/source/configuration_file.rst [hunk 10]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
...
+    }
+
+
+.. jsonschema::
+
+    {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "title": "project_name",
+        "$$description": [
+            "As the key suggests, this defines the name of the current project to which the groups and runs will be synced."
```
Comment (kanishk16) [inline docs/source/configuration_file.rst:196]: Just start with Defines to be in coherence with existing doc language

**Review by kanishk16 — APPROVED** [review]
Minor nitpicks... Apart from that LGTM!

File: docs/source/configuration_file.rst [hunk 11]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
     }
 
 
+Weights and Biases (WandB)
+-----------------
```
[the remark this answers is not in the document]
Comment (naga-karthik) [reply docs/source/configuration_file.rst:157]: You mean, with the hyphens? like extend them until after "(WandB)"?

File: docs/source/configuration_file.rst [hunk 12]
```diff
@@ -153,6 +153,134 @@ When possible, the folder name will follow the following convention:
...
+
+``ivadomed`` provides an additional option to track your ML experiments with the help of WandB. It provides a
```
[the remark this answers is not in the document]
Comment (naga-karthik) [reply docs/source/configuration_file.rst:159]: Aha! Good catch about the reference! Will update this

File: docs/source/configuration_file.rst (hunk 2 above)
Comment (naga-karthik) [reply docs/source/configuration_file.rst:280]: Oh yes, `debugging` it is!

**Review by kiristern — APPROVED** [review]

**Pull request merged**